src/site/content/en/blog/asyncify/index.md

```diff
@@ -0,0 +1,513 @@
+seamlessly restore the call stack and state and continue execution as if nothing has happened.
+
+Since any function in the module might make an asynchronous call, all the exports become potentially
+asynchronous too, so they get wrapped as well. You might have noticed in the example above that you
+need to `await` the result of `instance.exports.main()` to know when the execution is truly
+finished.
+
+### How does this all work under the hood?
+
+When Asyncify detects a call to one of the `ASYNCIFY_IMPORTS` functions, it starts an asynchronous
+operation, saves the entire state of the application, including the call stack and any temporary
+locals, and later, when that operation is finished, restores all the memory and call stack and
+resumes from the same place and with the same state as if the program has never stopped.
+
+This is quite similar to async-await feature in JavaScript that I showed earlier, but, unlike the
+JavaScript one, doesn't require any special syntax or runtime support from the language, and instead
+works by transforming plain synchronous functions at compile-time.
+
+When compiling the earlier shown asynchronous sleep example:
+
+```js
+puts("A");
+async_sleep(1);
+puts("B");
+```
+
+Asyncify takes this code and transforms it to roughly like the following one (pseudo-code, real
+transformation is more involved than this):
+
+```js
+if (mode == NORMAL_EXECUTION) {
+    puts("A");
+    async_sleep(1);
+    saveLocals();
+    mode = UNWINDING;
+    return;
+}
+if (mode == REWINDING) {
+    restoreLocals();
+    mode = NORMAL_EXECUTION;
+}
+puts("B");
+```
+
+Initially `mode` is set to `NORMAL_EXECUTION`. Correspondingly, the first time such transformed code
+is executed, only the part leading up to `async_sleep()` will get evaluated. As soon as the
+asynchronous operation is scheduled, Asyncify saves all the locals, and unwinds the stack by
+returning from each function all the way to the top, this way giving control back to the browser
+event loop.
+
+Then, once `async_sleep()` resolves, Asyncify support code will change `mode` to `REWINDING`, and
+call the function again. This time, the "normal execution" branch is skipped - since it already did
+the job last time and I want to avoid printing "A" twice - and instead it comes straight to the
+"rewinding" branch. Once it's reached, it restores all the stored locals, changes mode back to
+"normal" and continues the execution as if the code were never stopped in the first place.
+
+### Transformation costs
+
+Unfortunately, Asyncify transform isn't completely free, since it has to inject quite a bit of
+supporting code for storing and restoring all those locals, navigating the call stack under
+different modes and so on. It tries to modify only functions marked as asynchronous on the command
+line, as well as any of their potential callers, but that still might result in approximately 50 percent overhead to the
+uncompressed code size.
```

Comment:
@kaycebasques Just one more attempt, if this doesn't work either, I'll leave it as-is. What do you think?

```suggestion
different modes and so on. It tries to modify only functions marked as asynchronous on the command
line, as well as any of their potential callers, but the code size overhead still adds up to approximately 50% before compression.
```